### PR TITLE
[Task #960] Add flipper setup to the template

### PR DESCRIPTION
**Pull request by Benaaaaa**
## Summary

Introduce a feature flag management system to the rails template

**Related issue**:  [#960](https://app.productive.io/1-infinum/projects/2274/tasks/task/7909978?board=599220&filter=ODkzODc4)

### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

Add flipper gem to the default rails template and setup it's config

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

**Review by vr4b4c — COMMENTED** [review]
I'm getting following error
```
       rails        generate rspec:install
/Users/vrabac/.rbenv/versions/3.4.1/lib/ruby/gems/3.4.0/gems/flipper-active_record-1.3.4/lib/flipper/adapters/active_record.rb:292:in 'Flipper::Adapters::ActiveRecord#value_not_text?': undefined method 'column_for_attribute' for class Flipper::Gate (NoMethodError)

        @gate_class.column_for_attribute(:value).type != :text
```

File: recipes/run_generators.rb [hunk 1]
```diff
@@ -14,4 +14,6 @@
 end
 append_to_file 'lib/tasks/auto_annotate_models.rake', '# rubocop:enable Metrics/BlockLength, Rails/RakeEnvironment'
 rails_command 'generate strong_migrations:install'
+run 'rails g flipper:setup' if flipper_storage_adapter == FlipperStorageAdapters::ACTIVE_RECORD
+copy_file 'config/initializers/flipper.rb'
```
[the remark this answers is not in the document]
Comment (Benaaaaa) [reply recipes/run_generators.rb:18]: What if we do not have the db setup? rails g flipper:setup will fail 🤔 
It is reliant on the bin db:setup part being successful

**Comment by Benaaaaa** [comment]
> I'm getting following error

Could you give me more info on this error? I've tried recreating it locally with ruby 3.4.1 and it worker properly for me

**Comment by Benaaaaa** [comment]
Coud you tell me which rails version you used? The error you are reciving is connected to this issue: https://github.com/flippercloud/flipper/issues/917#issuecomment-2711141694, which should be fixed 🤔

**Comment by Benaaaaa** [comment]
So, I've went with the official docs implementation for the route instead of using the Flipper.instance in the config initializer. I've went through the flipper source code, and it just default to Flipper in case no argument is provided, which still had flipper flags correctly working when testing it out

File: recipes/run_generators.rb [hunk 2]
```diff
@@ -14,4 +14,6 @@
 end
 append_to_file 'lib/tasks/auto_annotate_models.rake', '# rubocop:enable Metrics/BlockLength, Rails/RakeEnvironment'
 rails_command 'generate strong_migrations:install'
+run 'rails g flipper:setup' if flipper_storage_adapter == FlipperStorageAdapters::ACTIVE_RECORD
```
Comment (vr4b4c) [inline recipes/run_generators.rb:17]: Should we run migrations here?
```suggestion
if flipper_storage_adapter == FlipperStorageAdapters::ACTIVE_RECORD
  run 'rails g flipper:setup'
  if in_docker?
    run 'rails db:migrate'
  else
    run "docker compose run --rm runner bash -c 'bundle exec rails db:migrate'"
  end
end
```

**Review by vr4b4c — APPROVED** [review]
Overall looks great 🙇

File: recipes/run_generators.rb (hunk 2 above)
Comment (Benaaaaa) [reply recipes/run_generators.rb:17]: I do not think that running migrations should be connected to the flipper setup itself. It is true that currently the only situation where we are generating any migrations is in the flipper active record, but if we run the rake db:migrate while having no migrations, it should not break e.g schema file got generated with no migrations:
image

So I suggest we run migrations later on, at the end of the whole process
